Pod/Classes/XExtensionItem.h

```diff
@@ -96,10 +140,51 @@ typedef UIImage *(^XExtensionItemThumbnailProvidingBlock)(CGSize suggestedSize,
  */
 @property (nonatomic, copy) NSDictionary *userInfo;
 
+/**
+ Add entries from a dictionary-serializable custom object to this paramter object’s `userInfo` dictionary.
+ 
+ @param dictionarySerializable Object whose entries should be added to this paramter object’s `userInfo` dictionary.
+ */
+- (void)addEntriesToUserInfo:(id <XExtensionItemDictionarySerializing>)dictionarySerializable;
```

Comment:
I’m not tied to this, but my thinking was that it made sense to create a protocol in order to encourage third-party developers to implement classes like [this one](https://github.com/tumblr/XExtensionItem/blob/master/Pod/Classes/Custom/Tumblr/XExtensionItemTumblrParameters.h), that can both be turned _into_ a dictionary and converted _from_ a dictionary. ([here’s how it’d look on the other side](https://github.com/tumblr/XExtensionItem/blob/master/Pod/Classes/Custom/Tumblr/XExtensionItem%2BTumblr.m)).

But maybe this isn’t necessary, and developers can just implement it however they want. I guess I just don’t want people hardcoding strings in their apps, I’d rather them be using “official” classes that are provided by developers of the apps that know to look for those string keys.